bindings/gumjs/gumquickcloak.c

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (C) 2024 Francesco Tamagni <mrmacete@protonmail.ch>
+ *
+ * Licence: wxWindows Library Licence, Version 3.1
+ */
+
+#include "gumquickcloak.h"
+
+#include "gumquickmacros.h"
+
+#include <gum/gumcloak.h>
+
+GUMJS_DECLARE_FUNCTION (gumjs_cloak_add_thread)
+GUMJS_DECLARE_FUNCTION (gumjs_cloak_remove_thread)
+GUMJS_DECLARE_FUNCTION (gumjs_cloak_has_thread)
+
+GUMJS_DECLARE_FUNCTION (gumjs_cloak_add_range)
+GUMJS_DECLARE_FUNCTION (gumjs_cloak_remove_range)
+GUMJS_DECLARE_FUNCTION (gumjs_cloak_has_range_containing)
+GUMJS_DECLARE_FUNCTION (gumjs_cloak_clip_range)
+
+GUMJS_DECLARE_FUNCTION (gumjs_cloak_add_file_descriptor)
+GUMJS_DECLARE_FUNCTION (gumjs_cloak_remove_file_descriptor)
+GUMJS_DECLARE_FUNCTION (gumjs_cloak_has_file_descriptor)
+
+
+static const JSCFunctionListEntry gumjs_cloak_entries[] =
+{
+  JS_CFUNC_DEF ("addThread", 0, gumjs_cloak_add_thread),
+  JS_CFUNC_DEF ("removeThread", 0, gumjs_cloak_remove_thread),
+  JS_CFUNC_DEF ("hasThread", 0, gumjs_cloak_has_thread),
+
+  JS_CFUNC_DEF ("addRange", 0, gumjs_cloak_add_range),
+  JS_CFUNC_DEF ("removeRange", 0, gumjs_cloak_remove_range),
+  JS_CFUNC_DEF ("hasRangeContaining", 0, gumjs_cloak_has_range_containing),
+  JS_CFUNC_DEF ("clipRange", 0, gumjs_cloak_clip_range),
+
+  JS_CFUNC_DEF ("addFileDescriptor", 0, gumjs_cloak_add_file_descriptor),
+  JS_CFUNC_DEF ("removeFileDescriptor", 0, gumjs_cloak_remove_file_descriptor),
+  JS_CFUNC_DEF ("hasFileDescriptor", 0, gumjs_cloak_has_file_descriptor),
+};
+
+void
+_gum_quick_cloak_init (GumQuickCloak * self,
+                       JSValue ns,
+                       GumQuickCore * core)
+{
+  JSContext * ctx = core->ctx;
+  JSValue obj;
+
+  self->core = core;
+
+  _gum_quick_core_store_module_data (core, "cloak", self);
+
+  obj = JS_NewObject (ctx);
+  JS_SetPropertyFunctionList (ctx, obj, gumjs_cloak_entries,
+      G_N_ELEMENTS (gumjs_cloak_entries));
+  JS_DefinePropertyValueStr (ctx, ns, "Cloak", obj, JS_PROP_C_W_E);
+}
+
+void
+_gum_quick_cloak_dispose (GumQuickCloak * self)
+{
+}
+
+void
+_gum_quick_cloak_finalize (GumQuickCloak * self)
+{
+}
+
+GUMJS_DEFINE_FUNCTION (gumjs_cloak_add_thread)
+{
+  GumThreadId thread_id;
+
+  if (!_gum_quick_args_parse (args, "Z", &thread_id))
+    return JS_EXCEPTION;
+
+  gum_cloak_add_thread (thread_id);
+
+  return JS_UNDEFINED;
+}
+
+GUMJS_DEFINE_FUNCTION (gumjs_cloak_remove_thread)
+{
+  GumThreadId thread_id;
+
+  if (!_gum_quick_args_parse (args, "Z", &thread_id))
+    return JS_EXCEPTION;
+
+  gum_cloak_remove_thread (thread_id);
+
+  return JS_UNDEFINED;
+}
+
+GUMJS_DEFINE_FUNCTION (gumjs_cloak_has_thread)
+{
+  GumThreadId thread_id;
+
+  if (!_gum_quick_args_parse (args, "Z", &thread_id))
+    return JS_EXCEPTION;
+
+  return JS_NewBool (ctx, gum_cloak_has_thread (thread_id));
+}
+
+GUMJS_DEFINE_FUNCTION (gumjs_cloak_add_range)
+{
+  gpointer address;
+  gsize size;
+  GumMemoryRange range;
+
+  if (!_gum_quick_args_parse (args, "pZ", &address, &size))
```

Comment:
Would be good to use a range object at the public API level, for `addRange()` and `removeRange()`, like we do in `Stalker.exclude()`. The C/C++ code can be kept as is, and like `Stalker.exclude()` we can unwrap the object in `runtime/core.js` and call the underscore-prefixed implementation.

Reply:
done!